pkg/apis/certmanager/v1alpha1/types.go

```diff
@@ -202,6 +203,26 @@ type ACMEIssuerDNS01ProviderAzureDNS struct {
 	HostedZoneName string `json:"hostedZoneName"`
 }
 
+// TSGIG HMAC hashing codes
+type TSIGAlgorithm string
+
+const (
+	HmacMD5    TSIGAlgorithm = "HmacMD5"
+	HmacSHA1   TSIGAlgorithm = "HmacSHA1"
+	HmacSHA256 TSIGAlgorithm = "HmacSHA256"
+	HmacSHA512 TSIGAlgorithm = "HmacSHA512"
```

Comment:
Replace Hmac with HMAC (in both type names and values)

Reply:
Do users have to specify this? I guess it cannot be auto-detected?

Reply:
> Replace Hmac with HMAC (in both type names and values)

FYI `github.com/miekg/dns`  internally uses the syntax `HmacMD5` so in same places of the code we will see this string, like [here](https://github.com/splashx/cert-manager/blob/425334068025f3d08450ac4952c5263aed90e94b/pkg/issuer/acme/dns/rfc2136/rfc2136.go#L64) - but I guess that's an ok distinction.

> Do users have to specify this? I guess it cannot be auto-detected?

This can't be auto-detected - it's set in the DNS server and there is no way to retrieve that info via standard DNS query. The